session/session.go

```diff
@@ -363,52 +286,15 @@ func (s *Session) handleIncoming(ctx context.Context, rcv rcvFrom) {
 }
 
 func (s *Session) updateReceiveCounters(ctx context.Context, rcv rcvFrom) {
-	s.sw.RLock()
-
-	for _, c := range rcv.ks {
-		if s.unlockedIsWanted(c) {
-			s.srs.RecordUniqueBlock()
-		} else if s.sw.pastWants.Has(c) {
-			s.srs.RecordDuplicateBlock()
-		}
-	}
-
-	s.sw.RUnlock()
+	// Record unique vs duplicate blocks
+	s.sw.ForEachUniqDup(rcv.ks, s.srs.RecordUniqueBlock, s.srs.RecordDuplicateBlock)
```

Comment:
Hm. I'm not sure of a better way to do this but this is a _bit_ funky. But it's probably fine.